src/builtins/core/now.rs

```diff
@@ -1,198 +1,143 @@
 //! The Temporal Now component
 
-use crate::iso::IsoDateTime;
 use crate::provider::TimeZoneProvider;
-use crate::time::EpochNanoseconds;
+use crate::unix_time::EpochNanoseconds;
 use crate::TemporalResult;
-
-#[cfg(feature = "sys")]
-use alloc::string::String;
+use crate::{iso::IsoDateTime, TemporalError};
 
 use super::{
     calendar::Calendar, timezone::TimeZone, Instant, PlainDate, PlainDateTime, PlainTime,
     ZonedDateTime,
 };
 
-/// The Temporal Now object.
-pub struct Now;
+#[derive(Debug, Default)]
+pub struct NowBuilder {
+    clock: Option<EpochNanoseconds>,
+    zone: Option<TimeZone>,
+}
+
+impl NowBuilder {
+    pub fn with_system_nanoseconds(mut self, nanoseconds: EpochNanoseconds) -> Self {
+        self.clock = Some(nanoseconds);
+        self
+    }
+
+    pub fn with_system_zone(mut self, zone: TimeZone) -> Self {
+        self.zone = Some(zone);
+        self
+    }
+
+    pub fn build(self) -> Now {
+        Now {
+            clock: self.clock,
+            zone: self.zone.unwrap_or_default(),
+        }
+    }
+}
+
+#[derive(Debug)]
+#[non_exhaustive] // Not constructable by external means
```

Comment:
Remark: I didn't know it can be applied t structs, very interesting!
Question: Isn't it redundant, tho? I don't think it's possible for the user to construct it anyways since it has private fields.

Reply:
Oh! true lol I've been so use to `pub` fields :laughing: